### PR TITLE
Show full dataset of referrals per campaign

### PR DESCRIPTION
**Pull request by kelvin-chappell**
This now shows results for all platforms, so that the sum of all the referral clicks should equal the number of page views for the campaign.

### CODE REVIEW

**Comment by kelvin-chappell** [comment]
[Reporting: Referrals: Impressions and CTR per card](https://trello.com/c/SjjlWMKh/146-reporting-referrals-impressions-and-ctr-per-card)

File: app/repositories/CampaignReferralRepository.scala [hunk 1]
```diff
@@ -1,17 +1,85 @@
...
+  path: Option[String],
+  containerIndex: Option[Int],
+  containerName: Option[String],
+  cardIndex: Option[Int],
+  cardName: Option[String],
+  clicks: Long,
+  unparsedComponent: Option[String]
+)
+
+object CampaignReferralRow {
```
Comment (LATaylor-guardian) [inline app/repositories/CampaignReferralRepository.scala]: Can we move this under `models` in its own file? I think this is how the rest of the domain models are defined in this application.

**Comment by LATaylor-guardian** [comment]
👍

**Comment by janua** [comment]
:+1:

**Pull request merged**